node/tests/test_providers.py

```diff
@@ -30,6 +27,34 @@ async def test_minimal_git(
     )
 
 
+async def test_git(
+    flatpak_builder: FlatpakBuilder,
+    provider_factory_spec: ProviderFactorySpec,
+    node_version: int,
+    shared_datadir: Path,
+) -> None:
+    with ManifestGenerator() as gen:
+        await provider_factory_spec.generate_modules('git', gen, node_version)
+
+    flatpak_builder.build(
+        sources=itertools.chain(
+            gen.ordered_sources(),
+            [
+                {
+                    'type': 'dir',
+                    'path': str(shared_datadir / 'packages' / 'git' / 'subdir'),
+                    'dest': 'subdir',
+                }
+            ],
+        ),
+        commands=[
+            provider_factory_spec.install_command,
+            """node -e 'require("array-range");require("is-empty-object");require("is-number");require("person-lib");require("to-camel-case");require("to-capital-case");require("to-no-case");require("to-space-case");require("@flatpak-node-generator-tests/subdir").sayHello()'""",
```

Comment:
this is...a bit horrifying, you could pull out the JS into a top-level multiline string constant, or just stick it in the shared datadir with the `package.json` (we don't really do that anywhere else, but the other lines are also *much* shorter!).

Reply:
Agreed, moved to a top-level constant.